### ISSUE TITLE
shadowsocks2022 client incompatible with database64128/shadowsocks-go's server implementation

### ISSUE THREAD

**Issue by dyhkwong** [v2fly/v2ray-core#2777]
## What version of V2Ray are you using?

v5.12.0

## What's your scenario of using V2Ray?

browse the Internet

## What problems have you encountered?

Use v2ray as client and shadowsocks-go (v1.9.0) as server, open Chrome (or Firefox) and configure it to use local socks5 proxy and visit https://www.baidu.com, Chrome shows `net::ERR_INCOMPLETE_CHUNKED_ENCODING 200 (OK)`  (Firefox shows `NS_ERROR_NET_PARTIAL_TRANSFER`).

I can't find the exact way to reproduce. Some website resources are broken but others are okay. Some unit tests may reproduce the issue.

Using v2ray as client and sing-box/shadowsocks-rust as server is okay. Using sing-box as client and shadowsocks-go as server is also okay.

## What's your expectation?

web page loaded correctly

## Please attach your configuration here

**Server configuration:**

```json
{
    "servers": [
        {
            "name": "ss-2022",
            "listen": "127.0.0.1:20220",
            "protocol": "2022-blake3-aes-128-gcm",
            "enableTCP": true,
            "enableUDP": true,
            "mtu": 1500,
            "psk": "qQln3GlVCZi5iJUObJVNCw==",
            "uPSKStorePath": "./upsks.json"
        }
    ]
}
```
upsks.json
```json
{
    "Steve": "oE/s2z9Q8EWORAB8B3UCxw=="
}
```

**Client configuration:**

```json
{
  "log": {
    "error": {
      "level": "Debug",
      "type": "Console"
    }
  },
  "outbounds": [
    {
      "protocol": "shadowsocks2022",
      "settings": {
        "address": "127.0.0.1",
        "port": 20220,
        "method": "2022-blake3-aes-128-gcm",
        "psk": "oE/s2z9Q8EWORAB8B3UCxw==",
        "ipsk": [
          "qQln3GlVCZi5iJUObJVNCw=="
        ]
      }
    }
  ],
  "inbounds": [
    {
      "protocol": "socks",
      "settings": {
        "udpEnabled": true,
        "address": "127.0.0.1"
      },
      "port": 1080
    }
  ]
}
```

## Please attach error logs here

**Server error log:**

```console
2023-11-22T22:39:04.171+0800 INFO service/tcp.go:121 Started TCP relay service listener {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220"}
2023-11-22T22:39:04.177+0800 INFO service/udp_session.go:171 Started UDP session relay service listener {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220"}
2023-11-22T22:39:10.256+0800 INFO service/tcp.go:246 Two-way relay started {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "clientAddress": "127.0.0.1:58522", "username": "Steve", "targetAddress": "www.baidu.com:443", "client": "direct", "initialPayloadLength": 0}
2023-11-22T22:39:10.319+0800 INFO service/tcp.go:246 Two-way relay started {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "clientAddress": "127.0.0.1:58525", "username": "Steve", "targetAddress": "ocsp.globalsign.com:80", "client": "direct", "initialPayloadLength": 0}
2023-11-22T22:39:10.563+0800 INFO service/tcp.go:246 Two-way relay started {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "clientAddress": "127.0.0.1:58529", "username": "Steve", "targetAddress": "psstatic.cdn.bcebos.com:443", "client": "direct", "initialPayloadLength": 0}
2023-11-22T22:39:10.595+0800 WARN service/tcp.go:255 Two-way relay failed {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "clientAddress": "127.0.0.1:58522", "username": "Steve", "targetAddress": "www.baidu.com:443", "client": "direct", "nl2r": 1304, "nr2l": 99213, "error": "read tcp 127.0.0.1:20220->127.0.0.1:58522: wsarecv: An existing connection was forcibly closed by the remote host."}
2023-11-22T22:39:10.606+0800 INFO service/tcp.go:246 Two-way relay started {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "clientAddress": "127.0.0.1:58530", "username": "Steve", "targetAddress": "www.baidu.com:443", "client": "direct", "initialPayloadLength": 0}
2023-11-22T22:39:18.390+0800 INFO shadowsocks-go/main.go:98 Received exit signal {"signal": "interrupt"}
2023-11-22T22:39:18.390+0800 INFO service/service.go:193 Stopped service {"service": "credential manager"}
2023-11-22T22:39:18.390+0800 INFO service/service.go:193 Stopped service {"service": "TCP relay service for ss-2022"}
2023-11-22T22:39:18.390+0800 INFO service/udp_session.go:498 Finished receiving from serverConn {"server": "ss-2022", "listener": 0, "listenAddress": "127.0.0.1:20220", "packetsReceived": 0, "payloadBytesReceived": 0}
2023-11-22T22:39:18.390+0800 INFO service/service.go:193 Stopped service {"service": "UDP session relay service for ss-2022"}

```

**Client error log:**

```console
V2Ray 5.12.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.21.3 windows/amd64)
A unified platform for anti-censorship.
2023/11/22 22:39:05 [Debug] app/log: Logger started
2023/11/22 22:39:05 [Debug] app/proxyman/inbound: creating stream worker on 0.0.0.0:1080
2023/11/22 22:39:05 [Info] transport/internet/tcp: listening TCP on 0.0.0.0:1080
2023/11/22 22:39:05 [Info] transport/internet/udp: listening UDP on 0.0.0.0:1080
2023/11/22 22:39:05 [Warning] V2Ray 5.12.0 started
2023/11/22 22:39:10 [Info] [3942489358] proxy/socks: TCP Connect request to tcp:www.baidu.com:443
2023/11/22 22:39:10 [Warning] [3942489358] app/dispatcher: default route for tcp:www.baidu.com:443
2023/11/22 22:39:10 [Info] [3942489358] transport/internet/tcp: dialing TCP to tcp:127.0.0.1:20220
2023/11/22 22:39:10 [Info] [3942489358] proxy/shadowsocks2022: tunneling request to tcp:www.baidu.com:443 via TCP:ip:"\x7f\x00\x00\x01"
2023/11/22 22:39:10 [Info] [4017683927] proxy/socks: TCP Connect request to tcp:ocsp.globalsign.com:80
2023/11/22 22:39:10 [Warning] [4017683927] app/dispatcher: default route for tcp:ocsp.globalsign.com:80
2023/11/22 22:39:10 [Info] [4017683927] transport/internet/tcp: dialing TCP to tcp:127.0.0.1:20220
2023/11/22 22:39:10 [Info] [4017683927] proxy/shadowsocks2022: tunneling request to tcp:ocsp.globalsign.com:80 via TCP:ip:"\x7f\x00\x00\x01"
2023/11/22 22:39:10 [Info] [3942489358] app/proxyman/outbound: failed to process outbound traffic > proxy/shadowsocks2022: connection ends > cipher: message authentication failed
2023/11/22 22:39:10 [Info] [3942489358] app/proxyman/inbound: connection ends > proxy/socks: connection ends > proxy/socks: failed to transport all TCP response > io: read/write on closed pipe
2023/11/22 22:39:10 [Info] [290327102] proxy/socks: TCP Connect request to tcp:psstatic.cdn.bcebos.com:443
2023/11/22 22:39:10 [Warning] [290327102] app/dispatcher: default route for tcp:psstatic.cdn.bcebos.com:443
2023/11/22 22:39:10 [Info] [290327102] transport/internet/tcp: dialing TCP to tcp:127.0.0.1:20220
2023/11/22 22:39:10 [Info] [290327102] proxy/shadowsocks2022: tunneling request to tcp:psstatic.cdn.bcebos.com:443 via TCP:ip:"\x7f\x00\x00\x01"
2023/11/22 22:39:10 [Info] [2269511462] proxy/socks: TCP Connect request to tcp:www.baidu.com:443
2023/11/22 22:39:10 [Warning] [2269511462] app/dispatcher: default route for tcp:www.baidu.com:443
2023/11/22 22:39:10 [Info] [2269511462] transport/internet/tcp: dialing TCP to tcp:127.0.0.1:20220
2023/11/22 22:39:10 [Info] [2269511462] proxy/shadowsocks2022: tunneling request to tcp:www.baidu.com:443 via TCP:ip:"\x7f\x00\x00\x01"
2023/11/22 22:39:14 [Debug] app/log: Logger closing
```

**Comment by database64128** [comment]
https://github.com/xiaokangwang/v2ray-core-1/blob/master/common/crypto/chunk.go#L63

Shadowsocks 2022 TCP's payload length uses the full range of an uint16, so this could overflow. @xiaokangwang

**Comment by xiaokangwang** [comment]
This issue should have been fixed in v5.12.1 release.
